packages/explat-client-react-helpers/src/index.tsx

```diff
@@ -38,32 +55,41 @@ interface ExPlatClientReactHelpers {
 			experimentAssignment: ExperimentAssignment | null
 		) => JSX.Element;
 		name: string;
+		options?: Partial< UseExperimentOptions >;
 	} ) => JSX.Element;
 }
 
 export default function createExPlatClientReactHelpers(
 	exPlatClient: ExPlatClient
 ): ExPlatClientReactHelpers {
-	const useExperiment = ( experimentName: string ): [ boolean, ExperimentAssignment | null ] => {
+	const useExperiment = (
+		experimentName: string,
+		providedOptions: Partial< UseExperimentOptions > = {}
+	): [ boolean, ExperimentAssignment | null ] => {
+		const options: UseExperimentOptions = { ...defaultUseExperimentOptions, ...providedOptions };
+
 		const [ previousExperimentName ] = useState( experimentName );
 		const [ state, setState ] = useState< [ boolean, ExperimentAssignment | null ] >( [
 			true,
 			null,
 		] );
+		const hasStartedLoadingRef = useRef< boolean >( false );
 
 		useEffect( () => {
 			let isSubscribed = true;
-			exPlatClient.loadExperimentAssignment( experimentName ).then( ( experimentAssignment ) => {
-				if ( isSubscribed ) {
-					setState( [ false, experimentAssignment ] );
-				}
-			} );
+			if ( hasStartedLoadingRef.current === false && options.isEligible ) {
+				hasStartedLoadingRef.current = true;
+				exPlatClient.loadExperimentAssignment( experimentName ).then( ( experimentAssignment ) => {
+					if ( isSubscribed ) {
+						setState( [ false, experimentAssignment ] );
+					}
+				} );
+			}
```

Comment:
If I understand correctly, `isEligible: false` simply disables the experiment, doesn't load the assignment from server and makes `useExperiment` return `null` assignment.

Unlike `experimentName`, `isEligible` can change while the component is rendered, so we need to handle updates correctly?

There is a race condition:
1. render `useExperiment( 'name', { isEligible: true } )`. Network load will start. Hook returns `isLoading = true`, `assignment = null`.
2. rerender with `useExperiment( 'name', { isEligible: false } )` while the network load is still in progress. Hook returns `isLoading = false`, `assignment = null`.
3. Now the network load finishes loading, but because `isSubscribed` is `false`, the result will be ignored.
4. rerender with `useExperiment( 'name' , { isEligible: true } )`. Because `hasStartedLoading` is `true`, new network load will not start. State and the return value will forever be `isLoading = true` and `assignment = null`.

Removing the `hasStartedLoadingRef` flag could fix this. I don't see it optimizing anything.

Reply:
Wow nice! Yeah I had added the `hasStartedLoadingRef` thinking I was preventing reloading the assignment from a `experimentName` change and I guess I messed this up.